topics/icrc_X_multiple_canister_call.md

```diff
@@ -0,0 +1,237 @@
+# ICRC-X: Call batch canisters
+
+[![Status Badge](https://img.shields.io/badge/STATUS-DRAFT-ffcc00.svg)](#)
+[![Extension Badge](https://img.shields.io/badge/Extends-ICRC--25-ffcc222.svg)](./icrc_25_signer_interaction_standard.md)
+[![Extension Badge](https://img.shields.io/badge/Extends-ICRC--49-ffcc222.svg)](./icrc_49_call_canister.md)
+
+<!-- TOC -->
+
+- [ICRC-X: Call batch canisters](#icrc-x-call-batch-canisters)
+  - [Summary](#summary)
+  - [Processing](#processing)
+    - [Examples](#examples)
+      - [Successful Batch Call](#successful-batch-call)
+      - [Batch Call with Error](#batch-call-with-error)
+
+## Summary
+
+This Method can be used by the relying party to request batch call to 3rd party canister executed by the signer using the requested identity. In order to prevent misuse of this method all `icrcX_call_batch_canisters` requests are subject to user approval.
+
+There is two main parameter for this standard
+
+1. `mode` : The mode to execute transaction in sequence or in parallel
+2. `requests` : An array of canister call requests
+
+Unfortunately there is currently no mode to execute all transactions in the batch in one block. The transactions are handled as independent transactions that are executed in parallel or in sequence. However, ICRCX allows signer to receive approval for all transactions in the batch from the user with one approval action. 
+
+## Processing
+**How it works**
+1. Relying party requests a ICRCX batch transaction request to the signer. The request can be either `parallel` mode or `sequence` mode (see more details below). 
+
+2. The signer fetches consent messages and shows a warning to user to approve the batch transaction. 
+
+3. The signer submits canister calls to target canisters:
+- mode `parallel`: Execute all requests simultaneously, without waiting for any individual request to complete before starting the next. All transactions will be started. But it is not guaranteed that all transactions will succeed.
+- mode `sequence`: 
+  - Execute each request one after the other, ensuring that each request is completed before starting the next one. If any transaction fails, the execution of the batch will stop without executing remaining queued transactions. 
+  - Evaluate the result of processing the request:
+     1. If the result is a successful response, add the result to the batch response. Proceed with processing the next request in the batch.
```

Comment:
As mentioned in the WG call and also Discord, we can't see if the call itself is successful regarding canister logic e.g. success or error result. But we can see (in the certificate) if the call has successfully [received a reply](https://github.com/dfinity/agent-js/blob/main/packages/agent/src/polling/index.ts#L65) from the canister or if something went wrong in between. 

This reply can be any byte array, in practice this is always cbor that will be decoded by the agent-js actor with knowledge of the candid definition of the canister. But even decoded from cbor into js values, there is no clear indicator of success or error, the value that indicates success or error is up to every canisters implementation itself to decide. It could be a boolean, enum or number range that indicates success or error.

So all we can do within this limitation is to stop the chain of calls when:
- No reply has been received
- Something else went wrong in between e.g. IC could not be reached